proto/backend.proto

```diff
@@ -413,17 +396,63 @@ message MutationResponse {
   bytes object_bytes = 4;
 }
 
+//-----------------------------------------------------------------
+// ResourceConversion -- EXPERIMENTAL and is subject to change until 12.??
+//-----------------------------------------------------------------
+
+// ResourceConversion is a service that can be used to convert resources between versions
+// See: https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#webhook-request-and-response
+service ResourceConversion {
+  // Convert objects to a target version
+  rpc ConvertObjects(ConversionRequest) returns (ConversionResponse);
+}
+
+// GroupVersion represents the API group and version of a resource.
+message GroupVersion{
+  string group = 1;
+  string version = 2;
+}
+
+// RawObject contains a resource serialized into a byte array with a content type
+message RawObject {
+  // Raw is the serialized object
+  bytes raw = 1;
+  // ContentType is the media type of the raw object
+  string content_type = 2;
```

Comment:
i was thinking that content_type would most likely always be `application/json`, but this would allow additional encodings if needed in the future